### PR TITLE
Update AGP, Kotlin, Compose, Api level and other dependecies

### PR DESCRIPTION
**Pull request by yamal-alm**
### :goal_net: What's the goal?
Update AGP, Kotlin, Compose, Api level and other dependencies

### :construction: How do we do it?
- Kotlin to 1.9.23
- Use latest BOM at the moment: 2024.04.01
- Use corresponding kotlin compose compiler version according to [this table](https://developer.android.com/jetpack/androidx/releases/compose-kotlin): 1.5.12 in this case
- AGP 8.2.2
- Target and compile sdk to 34
- Java 17
- Update every other dependencies

### :blue_book: Documentation changes?
- [x] No docs to update nor create

### :test_tube: How can I test this?
- [x] Everything looks ok in demo app

### CODE REVIEW

File: .github/workflows/manual_release.yml [hunk 1]
```diff
@@ -12,6 +12,12 @@ jobs:
       - name: Checkout repo
         uses: actions/checkout@v2
 
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
```
Comment (yamal-alm) [inline .github/workflows/manual_release.yml]: ubuntu-latests uses Java 11 by default, we need to set 17 explicitly

File: app/build.gradle
Comment (yamal-alm) [inline app/build.gradle:1]: app module:
- target 34
- java 17
- use latest compose version through BOM
- other dependencies updates

File: app/build.gradle [hunk 2]
```diff
@@ -4,12 +4,13 @@ plugins {
 }
 
 android {
-    compileSdk 33
+    namespace "com.telefonica.tweaks.demo"
```
Comment (yamal-alm) [inline app/build.gradle:7]: new `namespace` declaration coming from AGP 8.2.2

File: build.gradle [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
     ext {
-        compose_compiler_version = '1.4.2'
-        kotlin_version = '1.8.10'
-        compose_version = '1.3.2'
-        compose_material_version = '1.4.0'
-        dagger_version = "2.43.2"
+        compose_compiler_version = '1.5.12'
+        compose_bom_version = '2024.04.01'
+        kotlin_version = '1.9.23'
+        dagger_version = "2.51.1"
```
Comment (yamal-alm) [inline build.gradle:7]: ideally we should use a toml catalog, but that is out of scope of this task

File: build.gradle [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
 buildscript {
     ext {
-        compose_compiler_version = '1.4.2'
-        kotlin_version = '1.8.10'
-        compose_version = '1.3.2'
-        compose_material_version = '1.4.0'
-        dagger_version = "2.43.2"
+        compose_compiler_version = '1.5.12'
+        compose_bom_version = '2024.04.01'
+        kotlin_version = '1.9.23'
```
Comment (yamal-alm) [inline build.gradle:6]: latest kotlin version

File: build.gradle [hunk 5]
```diff
@@ -1,11 +1,10 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
     ext {
-        compose_compiler_version = '1.4.2'
-        kotlin_version = '1.8.10'
-        compose_version = '1.3.2'
-        compose_material_version = '1.4.0'
-        dagger_version = "2.43.2"
+        compose_compiler_version = '1.5.12'
```
Comment (yamal-alm) [inline build.gradle:4]: compose kotlin compiler accoding to [this table](https://developer.android.com/jetpack/androidx/releases/compose-kotlin) based on kotlin version

File: build.gradle [hunk 6]
```diff
@@ -15,7 +14,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.4.2'
+        classpath 'com.android.tools.build:gradle:8.2.2'
```
Comment (yamal-alm) [inline build.gradle:17]: agp updated as well

File: library/build.gradle
Comment (yamal-alm) [inline library/build.gradle:1]: library module:
- target 34
- java 17
- use latest compose version through BOM
- other dependencies updates

Comment (yamal-alm) [inline library/build.gradle:1]: library module:
- target 34
- java 17
- use latest compose version through BOM
- Use ksp instead of kapt for Dagger
- other dependencies updates

File: build.gradle [hunk 7]
```diff
@@ -24,6 +23,7 @@ buildscript {
 plugins {
     id("io.gitlab.arturbosch.detekt").version("1.18.1")
     id 'io.github.gradle-nexus.publish-plugin' version '1.3.0' apply false
+    id("com.google.devtools.ksp") version "1.9.23-1.0.20" apply false
```
Comment (yamal-alm) [inline build.gradle]: using ksp instead of kapt

File: app/build.gradle
Comment (jdelga) [reply app/build.gradle:1]: Why `setup-java` version downgrade ?

Comment (yamal-alm) [reply app/build.gradle:1]: where is the downgrade?

Comment (jdelga) [reply app/build.gradle:1]: Sorry, it was not in this line, but in the file above: https://github.com/Telefonica/tweaks/pull/40/commits/32f8059764ab94320322908373a3ed613636f6f4#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fL11

This also makes me think if `v3` can be used in all steps instead of `v2`

Comment (yamal-alm) [reply app/build.gradle:1]: I see, sure, I copy pasted the whole setup-java block from another repo. Let's use v3, thanks 👍

**Review by jdelga — APPROVED** [review]
Great update!

**Review by pmartinbTEF — APPROVED** [review]
Great job!
